clang-extract testsuite fails if gcc-14 is the system compiler

```
[   62s] 154/162 closure-1.c                ERROR           0.05s   exit status 99
[   62s] >>> MSAN_OPTIONS=halt_on_error=1:abort_on_error=1:print_summary=1:print_stacktrace=1 ASAN_OPTIONS=halt_on_error=1:abort_on_error=1:print_summary=1 MESON_TEST_ITERATION=1 UBSAN_OPTIONS=halt_on_error=1:abort_on_error=1:print_summary=1:print_stacktrace=1 '/home/abuild/rpmbuild/BUILD/clang-extract-0~20240722.a998e91/testsuite/lib/runtest.py' -inline-test '/home/abuild/rpmbuild/BUILD/clang-extract-0~20240722.a998e91/testsuite/inline/closure-1.c' -o '/home/abuild/rpmbuild/BUILD/clang-extract-0~20240722.a998e91/x86_64-suse-linux/testsuite/inline/closure-1.c.log'
[   62s] ――――――――――――――――――――――――――――――――――――― ✀  ―――――――――――――――――――――――――――――――――――――
[   62s] /home/abuild/rpmbuild/BUILD/clang-extract-0~20240722.a998e91/testsuite/inline/closure-1.c:9:34: error: return type defaults to ‘int’ [-Wimplicit-int]
[   62s]     9 | static __attribute__((noinline)) h(void)
[   62s]       |                                  ^
[   62s] 
[   62s] ――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――
```
Adjust the testcase so it silence this error.

--- a/testsuite/inline/closure-1.c
+++ b/testsuite/inline/closure-1.c
@@ -1,4 +1,4 @@
-/* { dg-compile "-fdump-ipa-clones -O3 -g3"} */
+/* { dg-compile "-fdump-ipa-clones -O3 -g3 -Wno-implicit-int"} */
 /* { dg-options "-compute-closure main"} */
 
 static inline int g(void)

